### PR TITLE
Feature/volunteer - Volunteer Component

### PR DESCRIPTION
**Pull request by kevy7**
Issue: https://github.com/kdv24/Jr-Developer-Site/projects/1#card-22640271

Built a Volunteer form as requested from the issue above. I built the form using formspree for now, since we have yet to deploy our website using netlify. With formspree, the email placed in the action attribute of the form will be receiving form submission from users. i.e., see example below.
`<form action="https://formspree.io/anyEmailHere@email.com" method="POST">`
I used css modules to style the volunteer component. The styling is simple for now and can be updated later once we get more advice from designers.

Please check if the coding for this component looks good. Please review the volunteer.module.css file as well and see if my naming conventions and styling looks good for now.

### CODE REVIEW

File: src/components/volunteer/volunteer.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                    <input type="text" name="name" id="userName" onChange={this.handleInputChanges} value={this.state.name} />
+                        
+
+                    <label htmlFor="email">Email:</label>
+                    <input type="text" name="email" id="email" onChange={this.handleInputChanges} value={this.state.email} />
+
+                    <label htmlFor="volunteerList" >Volunteer Options: </label>
+                    <select name="volunteerList" id="volunteerList" value={this.state.volunteerList} onBlur={this.handleInputChanges} onChange={this.handleInputChanges}>
+                        <option value="twitterManagement" >Twitter Account Management</option>
+                        <option value="greeter" >Guess Greeter</option>
```
Comment (kdv24) [inline src/components/volunteer/volunteer.js]: I'm guessing you mean "Guest" not "Guess" greeter? :smiley:

**Comment by kdv24** [comment]
@kevy7 this looks good! Your PR itself is really well written, and made it easy for me (and anyone else who's jumping in to see the work) to understand what's been done and why.  Since I've never used formspree before, I especially appreciated your description of the form action with formspree.

Styling looks good for now- we can adjust once we have wireframes back from Keith.  Thanks for making it responsive too.  I tested it on mobile and we may want to adjust some padding/margin stuff once we have an actual design we're following, but until then this is great!

As for naming conventions, what I've seen in the React documentation is that files are named with capital letters.  Here's an example from a page on file structure.  
Screen Shot 2019-08-07 at 12 16 11 PM

I've worked on codebases before where the components that are classes are capitalized and others are not. I've also seen some that are all lowercase.  My preference is that we use capitals, but I'm open to discussion.  Currently the 'aboutUs' section is not capitals, but I thought we'd changed that in the last PR to 'AboutUs.'  Thoughts?

**Comment by kdv24** [comment]
It looks like you may need to rebase to our current master to resolve the conflict in `index`.  Let me know if you have questions on that.

Great work! 🎉

File: src/components/volunteer/volunteer.js (hunk 1 above)
Comment (kevy7) [reply src/components/volunteer/volunteer.js]: Woops! just updated this to be "Guest" instead of "Guess". Thanks for catching that!

**Comment by kevy7** [comment]
Thanks for the review @kdv24. For the aboutUs component, I capitalized it later on with the previous pull request that I made but we're no longer using that component. Molly and I decided to use the expandable-section component instead because it can be re-used for the about us and code of conduct section. I agree that we should be capitalizing our components and will do so moving forward. As for the naming convention for files, I'm okay with capitalizing files as well since it's suggested in the React Documentation. Let me know what you think! FYI, sorry for all of the commits. I had issues with my rebase but it finally worked!

**Comment by kdv24** [comment]
Don't worry about all the commits, but if you'd like, I'm also happy to work with you to go through it and squash some of them. Personally, I like being able to do that with my branches because it lets me clean things up some.  Some people prefer not to- either is fine.

Sounds good on the naming conventions. 

I'm happy to go ahead and merge this now so we can get the code in there, or we can do the rebasing/squashing if you want.  Either way is totally good with me.  If you want, feel free to merge it yourself or let me know, and I'll do it. 

Thanks @kevy7 !

**Comment by kevy7** [comment]
@kdv24 Sorry didn't realized you commented on this. I'm going to merge it for now but yes, I would love to go through rebasing with you!

**Pull request merged**